### ISSUE TITLE
Whitelist for disabling on sites that it breaks

### ISSUE THREAD

**Issue by newhouse** [newhouse/url-tracking-stripper#40]
User commented in Chrome Webstore: 
> Great addon, but really needs a whitelist so you can disable it for sites it breaks.

**Comment by JokerQyou** [comment]
Either a whitelist to disable stripping or a blacklist to enable stripping (limit stripping to given list of websites) would be great.
Currently I'd like to report usability issue with GitLab installations. When visiting a GitLab repo (e.g. ![https://gitlab.com/gitlab-org/gitlab-ce]), I cannot toggle between branches, because this extension would strip away the `ref` parameter in URL, causing a 404 error. The `ref` parameter (`reference`) comes from `git` ecosystem and is actively used by various git-related websites.

**Comment by JokerQyou** [comment]
For your reference, I found this Firefox addon on Github which seems to support custom rules when stripping tracking parameters, [link here](https://github.com/Smile4ever/firefoxaddons/tree/master/Neat%20URL-webext). And I think its format is somewhat well defined. Maybe we could support this format? Or create our own.

**Comment by newhouse** [comment]
OK, thanks @JokerQyou I've been meaning to dive into this and I'll try to get to it this weekend if possible!

**Comment by mikhoul** [comment]
@newhouse  I agree a "**toggle switch**" would be useful to enable/disable the extension for testing in conjunction with an "**option for whitelisting/blacklisting**" sites.

It would be really useful 😊

Regards :octocat:

**Comment by dy-dx** [comment]
`?ref` is also used by the Steam support website:

https://support.steampowered.com/kb_article.php?ref=1558-AFCM-4577

**Comment by iczman** [comment]
+1, this is breaking GitLab's branch switching feature. I had to disable this extension in order to continue my work. Would be great if it support some kind of whitelist or blacklist.
